File: solutions/scala/bob/1/src/main/scala/Bob.scala
object Bob {
  def isShout(s: String): Boolean = s == s.toUpperCase && s.exists(_.isLetter)

  def isQuestion(s: String): Boolean = s.trim.endsWith("?")

  def isSilent(s: String): Boolean = s.trim.isEmpty

  def response(s: String): String = (isSilent(s), isShout(s), isQuestion(s)) match {
    case (true, _, _) => "Fine. Be that way!"
    case (_, true, true) => "Calm down, I know what I'm doing!"
    case (_, true, _) => "Whoa, chill out!"
    case (_, _, true) => "Sure."
    case _ => "Whatever."
  }
}

File: solutions/java/bob/1/src/main/java/Bob.java
class Bob {

    String hey(String input) {
        return switch (input.trim()) {
            case String s when s.isEmpty() -> "Fine. Be that way!";
            case String s when s.matches("(?=.*[A-Z])(?!.*[a-z]).*\\?") -> "Calm down, I know what I'm doing!";
            case String s when s.endsWith("?") -> "Sure.";
            case String s when s.matches("(?=.*[A-Z])(?!.*[a-z]).*") -> "Whoa, chill out!";
            default -> "Whatever.";
        };
    }
}